### PR TITLE
chore(ci): properly install recent version of git (needed >= 2.18 by checkout action)

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area CI

**What this PR does / why we need it**:

Moreover, fixed path to output packages from Falco `make package` target.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/reusable_build_packages.yaml [hunk 1]
```diff
@@ -112,21 +114,21 @@ jobs:
         with:
           name: falco-${{ steps.store_version.outputs.version }}-${{ inputs.arch }}.tar.gz
           path: |
-            ${{ github.workspace }}/build/packages/falco-*.tar.gz
+            ${{ github.workspace }}/build/falco-*.tar.gz
```
Comment (FedeDP) [inline .github/workflows/reusable_build_packages.yaml:117]: Fixed path.

**Review by Andreagit97 — APPROVED** [review]
/approve

**Comment by FedeDP** [comment]
Further fixes on top of #2501 .

**Review by jasondellaluce — APPROVED** [review]

**Comment by FedeDP** [comment]
/unhold

Lets try!

**Pull request merged**